### PR TITLE
Compile for more platforms on release

### PR DESCRIPTION
**Pull request by 6543**
- [x] move docker stuff into own pipeline
 - [x] more platforms on release

### CODE REVIEW

File: .woodpecker/docker.yml [hunk 1]
```diff
@@ -0,0 +1,200 @@
+pipeline:
+  build-web:
+    image: node:16-alpine
+    commands:
+      - cd web/
+      - yarn install --frozen-lockfile
+      - yarn build
```
Comment (6543) [inline .woodpecker/docker.yml:7]: @anbraten I wonder - do there exist flags for "release mode"  to optimize build or is this default?

Comment (anbraten) [reply .woodpecker/docker.yml:7]: Build is already the production build with optimization. `yarn start` does the hot-reload dev build.

Comment (anbraten) [reply .woodpecker/docker.yml:7]: We should definitely think about caching for the web builds `dist` folder & `node_modules`

File: .woodpecker/docker.yml [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+
+  ###############
+  # S e r v e r #
+  ###############
+
+  # needed until https://github.com/woodpecker-ci/woodpecker/pull/635
```
Comment (anbraten) [inline .woodpecker/docker.yml:13]:
```suggestion
  # TODO: needed until https://github.com/woodpecker-ci/woodpecker/pull/635
```

File: .woodpecker/docker.yml [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+  build-server:
```
Comment (anbraten) [inline .woodpecker/docker.yml:14]: We are currently missing `publish-server-dryrun`, `publish-server-alpine-dryrun`, `publish-agent-alpine-dryrun` and `publish-cli-alpine-dryrun`

**Review by anbraten — APPROVED** [review]

File: .woodpecker/docker.yml [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+    settings:
+      repo: woodpeckerci/woodpecker-agent
+      dockerfile: docker/Dockerfile.agent.multiarch
+      platforms: linux/arm/v7,linux/arm64/v8,linux/amd64,linux/ppc64le,windows/amd64,darwin/amd64,darwin/arm64,freebsd/arm64,freebsd/amd64,openbsd/arm64,openbsd/amd64
+      tag: [latest, "${CI_COMMIT_TAG}"]
+    when:
+      event: tag
+
+  release-agent-alpine:
+    group: bundle
```
Comment (6543) [inline .woodpecker/docker.yml:121]:
```suggestion
    group: docker
```

File: .woodpecker/docker.yml [hunk 5]
```diff
@@ -0,0 +1,200 @@
...
+      repo: woodpeckerci/woodpecker-agent
+      dockerfile: docker/Dockerfile.agent.alpine.multiarch
+      platforms: linux/arm/v7,linux/arm64/v8,linux/amd64,linux/ppc64le
+      tag: next-alpine
+    when:
+      branch: ${CI_REPO_DEFAULT_BRANCH}
+      event: push
+
+  release-agent:
+    group: bundle
```
Comment (6543) [inline .woodpecker/docker.yml:109]:
```suggestion
    group: docker
```

File: .woodpecker/docker.yml [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+      repo: woodpeckerci/woodpecker-agent
+      dockerfile: docker/Dockerfile.agent.multiarch
+      platforms: linux/arm/v7,linux/arm64/v8,linux/amd64,linux/ppc64le,windows/amd64,darwin/amd64,darwin/arm64,freebsd/arm64,freebsd/amd64,openbsd/arm64,openbsd/amd64
+      tag: next
+    when:
+      branch: ${CI_REPO_DEFAULT_BRANCH}
+      event: push
+
+  publish-agent-alpine:
+    group: bundle
```
Comment (6543) [inline .woodpecker/docker.yml:96]:
```suggestion
    group: docker
```

File: .woodpecker/docker.yml [hunk 7]
```diff
@@ -0,0 +1,200 @@
...
+      dry_run: true
+      repo: woodpeckerci/woodpecker-agent
+      dockerfile: docker/Dockerfile.agent.multiarch
+      platforms: linux/arm/v7,linux/arm64/v8,linux/amd64,linux/ppc64le,windows/amd64,darwin/amd64,darwin/arm64
+      tag: test
+    when:
+      event: pull_request
+
+  publish-agent:
+    group: bundle
```
Comment (6543) [inline .woodpecker/docker.yml:83]:
```suggestion
    group: docker
```

File: .woodpecker/docker.yml [hunk 8]
```diff
@@ -0,0 +1,200 @@
...
+      tag: [latest-alpine, "${CI_COMMIT_TAG}-alpine"]
+    when:
+      event: tag
+
+  #############
+  # A g e n t #
+  #############
+
+  publish-agent-dryrun:
+    group: bundle
```
Comment (6543) [inline .woodpecker/docker.yml:71]:
```suggestion
    group: docker
```

File: .woodpecker/docker.yml [hunk 9]
```diff
@@ -0,0 +1,200 @@
...
+    secrets: [docker_username, docker_password]
+    settings:
+      repo: woodpeckerci/woodpecker-server
+      dockerfile: docker/Dockerfile.server
+      tag: [latest, "${CI_COMMIT_TAG}"]
+    when:
+      event: tag
+
+  release-server-alpine:
+    group: bundle
```
Comment (6543) [inline .woodpecker/docker.yml:56]:
```suggestion
    group: docker
```

File: .woodpecker/docker.yml [hunk 10]
```diff
@@ -0,0 +1,200 @@
...
+    settings:
+      repo: woodpeckerci/woodpecker-server
+      dockerfile: docker/Dockerfile.server.alpine
+      tag: next-alpine
+    when:
+      branch: ${CI_REPO_DEFAULT_BRANCH}
+      event: push
+
+  release-server:
+    group: bundle
```
Comment (6543) [inline .woodpecker/docker.yml:45]:
```suggestion
    group: docker
```

File: .woodpecker/docker.yml [hunk 11]
```diff
@@ -0,0 +1,200 @@
...
+      repo: woodpeckerci/woodpecker-server
+      dockerfile: docker/Dockerfile.server
+      tag: next
+    when:
+      branch: ${CI_REPO_DEFAULT_BRANCH}
+      event: push
+
+  publish-server-alpine:
+    image: plugins/docker
+    group: bundle
```
Comment (6543) [inline .woodpecker/docker.yml:34]:
```suggestion
    group: docker
```

File: .woodpecker/docker.yml [hunk 12]
```diff
@@ -0,0 +1,200 @@
...
+  # needed until https://github.com/woodpecker-ci/woodpecker/pull/635
+  build-server:
+    group: build
+    image: golang:1.16
+    commands:
+      - make release-server
+
+  publish-server:
+    image: plugins/docker
+    group: bundle
```
Comment (6543) [inline .woodpecker/docker.yml:22]:
```suggestion
    group: docker
```

File: .woodpecker/docker.yml [hunk 13]
```diff
@@ -0,0 +1,200 @@
...
+      - yarn install --frozen-lockfile
+      - yarn build
+
+  ###############
+  # S e r v e r #
+  ###############
+
+  # needed until https://github.com/woodpecker-ci/woodpecker/pull/635
+  build-server:
+    group: build
```
Comment (6543) [inline .woodpecker/docker.yml:15]:
```suggestion
```

File: .woodpecker/docker.yml (hunk 3 above)
Comment (6543) [reply .woodpecker/docker.yml:14]: no only dryrun for server is missing ...

Comment (6543) [reply .woodpecker/docker.yml:14]: ... I'd like to fix this in #635 ...

Comment (anbraten) [reply .woodpecker/docker.yml:14]: But we do not have a dryrun for `publish-agent-alpine` atm? 🤔

Comment (6543) [reply .woodpecker/docker.yml:14]: ... I'll add it now - will be changed later

**Pull request merged**